staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go

```diff
@@ -157,9 +172,16 @@ func TestGracefulTerminationWithKeepListeningDuringGracefulTerminationDisabled(t
 	steps := func(before bool, name string, e lifecycleSignal) {
 		// Before AfterShutdownDelayDuration event is signaled, the test
 		// will send request(s) to assert on expected behavior.
-		if name == "AfterShutdownDelayDuration" && before {
+		switch {
+		case name == "AfterShutdownDelayDuration" && before:
 			// it unblocks the verification step and waits for it to complete
 			<-delayedStopVerificationStep.done()
+		case name == "AfterShutdownDelayDuration" && !before:
+			// we want to inject a delay in between AfterShutdownDelayDuration
+			// and when the HandlerChainWaitGroup.Wait() is invoked, this provides
+			// the pre shutdown hook a flake free time window in which it can exercise
+			// an API call without an error.
+			<-time.After(time.Second)
```

Comment:
Do we actually use it for anything in this test?

Your pre-shutdown hook is reacting on `preShutdownHookBlockedCh` anyway, so I'm not sure why this is needed.

Reply:
I did not need it at the end